bin/hummingbot.py

```diff
@@ -72,8 +76,48 @@ async def main():
         await safe_gather(*tasks)
 
 
+def start_docker(queue, event):
+    docker_client = docker.APIClient(base_url='unix://var/run/docker.sock')
+    while True:
+        try:
+            try:
+                method, kwargs = queue.recv()
+            except Exception:
+                break
+            if isinstance(kwargs, list):
+                response = getattr(docker_client, method)(kwargs[0], **kwargs[1])
+            else:
+                response = getattr(docker_client, method)(**kwargs)
+
+            if isinstance(response, types.GeneratorType):
+                event.set()
+                for stream in response:
+                    queue.send(stream)
+                queue.send(None)
+                event.clear()
+            else:
+                queue.send(response)
+        except Exception as e:
+            queue.send(e)
+
+
 if __name__ == "__main__":
+    # IPC pipe
+    client, dock = aioprocessing.AioPipe()
+    event = aioprocessing.AioEvent()
+
+    # fork app
+    p = Process(target=start_docker, args=(client, event,))
+    p.start()
+
     chdir_to_data_directory()
     if login_prompt():
         ev_loop: asyncio.AbstractEventLoop = asyncio.get_event_loop()
-        ev_loop.run_until_complete(main())
+        ev_loop.run_until_complete(main(dock, event))
+
+    # stop ipc
+    dock.send(None)
```

Comment:
These are clean up instructions, and should be run even if the stuff above errors out for any reason. So you should use a `try ... finally ...` block to do the clean up.